src/main/kotlin/io/emeraldpay/dshackle/upstream/grpc/BitcoinGrpcUpstream.kt

```diff
@@ -149,9 +149,12 @@ class BitcoinGrpcUpstream(
     override fun stop() {
     }
 
-    override fun update(conf: BlockchainOuterClass.DescribeChain) {
-        upstreamStatus.update(conf)
-        this.capabilities = RemoteCapabilities.extract(conf)
+    override fun update(conf: BlockchainOuterClass.DescribeChain): Boolean {
+        var changed = false
+        changed = changed || upstreamStatus.update(conf)
+        val newCapabilities = RemoteCapabilities.extract(conf)
+        changed = changed || (newCapabilities != capabilities)
         conf.status?.let { status -> onStatus(status) }
+        return changed
```

Comment:
May be `upstreamStatus.update(conf) || (newCapabilities != capabilities)` at once instead of var?